File: banchus-common/src/main/java/pe/nanamochi/banchus/database/repository/BeatmapRepository.java
package pe.nanamochi.banchus.database.repository;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import pe.nanamochi.banchus.database.entity.Beatmap;

@Repository
public interface BeatmapRepository extends JpaRepository<Beatmap, Integer> {
  Optional<Beatmap> findById(int id);

  Optional<Beatmap> findByMd5(String md5);
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/database/repository/BeatmapRepository.kt
package pe.nanamochi.banchus.database.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pe.nanamochi.banchus.database.entity.Beatmap

@Repository
interface BeatmapRepository : JpaRepository<Beatmap, Int> {
    fun findBeatmapById(id: Int): Beatmap?

    fun findByMd5(md5: String): Beatmap?
}
